### PR TITLE
[SD-2740] Ensure that all items in a package validate and are not loc…

### PR DESCRIPTION
**Pull request by marwoodandrew**
…ked on publish

### CODE REVIEW

**Comment by amagdas** [comment]
:+1:  Needs rebasing

**Comment by mugurrus** [comment]
:+1:

**Comment by akintolga** [comment]
:+1:

File: server/apps/publish/archive_publish.py [hunk 1]
```diff
@@ -92,6 +92,21 @@ def on_update(self, updates, original):
         if validation_errors[0]:
             raise ValidationError(validation_errors)
 
+        # We do not allow packages to be published if any items in the package do not validate
+        if original[ITEM_TYPE] == CONTENT_TYPE.COMPOSITE:
+            items = [ref.get('residRef') for group in original.get('groups', [])
+                     for ref in group.get('refs', []) if 'residRef' in ref]
+            if items:
+                for guid in items:
+                    doc = super().find_one(req=None, _id=guid)
```
Comment (mdhaman) [inline server/apps/publish/archive_publish.py]: I think we do the similar with line https://github.com/marwoodandrew/superdesk/blob/master/server/apps/publish/archive_publish.py#L89 i.e. original + updates for validating not composite items.

Comment (sivakuna-aap) [inline server/apps/publish/archive_publish.py]: It's already done in pr#901. https://github.com/superdesk/superdesk/blob/master/server/apps/publish/archive_publish.py#L89-L91

Comment (mdhaman) [inline server/apps/publish/archive_publish.py]: cool

**Comment by mdhaman** [comment]
:+1:  rebase needed

**Pull request merged**